tracing-attributes/src/expand.rs

```diff
@@ -49,8 +50,35 @@ pub(crate) fn gen_function<'a, B: ToTokens + 'a>(
 
     let warnings = args.warnings();
 
+    let block = if let ReturnType::Type(_, return_type) = &output {
+        let return_type = erase_impl_trait(return_type);
+        // Install a fake return statement as the first thing in the function
+        // body, so that we eagerly infer that the return type is what we
+        // declared in the async fn signature.
+        let fake_return_edge = quote_spanned! {return_type.span()=>
+            #[allow(unreachable_code)]
+            if false {
+                let __tracing_attr_fake_return: #return_type =
+                    panic!("this is just for type inference, and is unreachable code");
+                return __tracing_attr_fake_return;
```

Comment:
these names probably don't _need_ to have the `__tracing_attr_`, since they're only defined inside the scope of the `if` and won't conflict with names defined in the user code? but it might be worth keeping so that it's clear where they came from...